### PR TITLE
CA-160971: Fix the Error Message to 'Remove the patch backup files' crea...

### PR DESCRIPTION
**Pull request by sharady**
...ted when patch-apply fails and User try to re-apply the patch

Signed-off-by: Sharad Yadav <sharad.yadav@citrix.com>

### CODE REVIEW

File: ocaml/idl/api_errors.ml [hunk 1]
```diff
@@ -335,6 +335,7 @@ let patch_precheck_failed_wrong_server_build = "PATCH_PRECHECK_FAILED_WRONG_SERV
 let patch_precheck_failed_vm_running = "PATCH_PRECHECK_FAILED_VM_RUNNING"
 let patch_precheck_tools_iso_mounted = "PATCH_PRECHECK_FAILED_ISO_MOUNTED"
 let patch_apply_failed = "PATCH_APPLY_FAILED"
+let patch_apply_failed_backup_files_exists = "PATCH_APPLY_FAILED_BACKUP_FILES_EXISTS"
```
Comment (euanh) [inline ocaml/idl/api_errors.ml]: This should probably be "patch_apply_failed_backup_files_exist"

File: ocaml/idl/datamodel.ml [hunk 2]
```diff
@@ -1091,6 +1091,8 @@ let _ =
     ~doc:"This patch has already been applied" ();
   error Api_errors.patch_apply_failed [ "output" ]
     ~doc:"The patch apply failed.  Please see attached output." ();
+  error Api_errors.patch_apply_failed_backup_files_exists [ "output" ]
+    ~doc:"The patch apply failed" ();
```
Comment (euanh) [inline ocaml/idl/datamodel.ml]: Should we not say why the patch application failed here?

File: ocaml/idl/api_errors.ml (hunk 1 above)
Comment (thomassa) [inline ocaml/idl/api_errors.ml]: Or `..._file_exists`?

File: ocaml/xapi/xapi_pool_patch.ml [hunk 3]
```diff
@@ -645,8 +645,18 @@ let apply ~__context ~self ~host =
 	        output;
 		
         | Failure(log, exn) ->
-	        debug "error from patch application: %s" log;
-	        raise (Api_errors.Server_error(Api_errors.patch_apply_failed, [log]))
+		debug "error from patch application: %s" log;
+           	let error_string = "Backup files already present - aborting." in
+           	if List.length (Xstringext.String.find_all error_string log) = 0 then
```
Comment (euanh) [inline ocaml/xapi/xapi_pool_patch.ml]: Searching for this string seems quite brittle - what happens if the patch tool's error message changes slightly?   Is there a better way to detect this failure?

**Comment by euanh** [comment]
@sharady: Grepping for that string is probably the best we can do.   It's a pity that the script doesn't have more machine-readable output.

**Pull request merged**